src/utils/options/normalizeInputOptions.ts

```diff
@@ -54,7 +55,7 @@ export function normalizeInputOptions(config: InputOptions): {
 		moduleContext: getModuleContext(config, context),
 		onwarn,
 		perf: config.perf || false,
-		plugins: ensureArray(config.plugins),
+		plugins: flatten(ensureArray(config.plugins)),
```

Comment:
Two things:
 - In the end, `flatten` just reimplements `[].flat(Infinity)`, which I would prefer to use
 - `ensureArray` is only called on the original array. However, it also filters falsy plugins, which should also happen in nested arrays.

```suggestion
		plugins: ensureArray(config.plugins).flat(Infinity).filter(Boolean),
```

Reply:
I just noticed `target` in `tsconfig` is `ES2020` now. I implemented this feature before 3.0 was released. Thanks for suggestions!

Reply:
Ah right, there was a release yesterday 😄